ext/dom/tests/bug79701/id_property.phpt

```diff
@@ -0,0 +1,53 @@
+--TEST--
+Bug #79701 (getElementById does not correctly work with duplicate definitions) - id property variation
+--EXTENSIONS--
+dom
+--FILE--
+<?php
+$dom = Dom\XMLDocument::createFromString(<<<XML
+<root>
+    <test1/>
+    <test2/>
+</root>
+XML);
+
+$test1 = $dom->documentElement->firstElementChild;
+$test2 = $test1->nextElementSibling;
+
+echo "--- After parsing ---\n";
+var_dump($dom->getElementById("x")?->nodeName);
+
+echo "--- After setting test2 ---\n";
+$test2->id = "x";
+var_dump($dom->getElementById("x")?->nodeName);
+echo "--- After setting test1 ---\n";
+$test1->id = "x";
```

Comment:
Ah brilliant, we can have two ids that are identical. Not going to question the behaviour as that's probably some libxml stuff.

Reply:
It's unfortunately allowed by the DOM spec. In that case, it should return the first element with that id.